src/pipeline.ml

```diff
@@ -162,21 +170,22 @@ module Cluster = struct
     | None -> None
 
   (* Build [src/dockerfile] as a Docker service. *)
-  let build { sched; dockerfile; options; archs } ?(additional_build_args=Current.return []) repo src : unit Current.t =
-    Current.component "HEADs" |>
+  let build { sched; dockerfile; options; archs } ?(additional_build_args=Current.return []) repo commit =
+    Current.component "HEADs %a" (Fmt.list Arch.pp) archs |>
     let** additional_build_args = additional_build_args in
     let options = { options with build_args = additional_build_args @ options.build_args } in
     let build_arch arch =
-      let* src = src in
-      let build = Current_ocluster.build sched ~options ~pool:(pool_id arch) ~src:(Current.return [src]) dockerfile in
-      let hash = Current_git.Commit_id.hash src in
-      let () = Logs.info (fun f -> f "Building arch: %s repo: %s hash: %s" (pool_id arch) (Fmt.str("%s/%s") repo.Github.Repo_id.owner repo.name) hash) in
-      let+ job_id = get_job_id build in
-      let job_str = match job_id with | Some x -> x | None -> "None" in
-      let () = Logs.info (fun f -> f "Recording repo: %s hash: %s job: %s" (Fmt.str("%s/%s") repo.owner repo.name) hash job_str) in
-      Index.record ~repo ~hash [("build", job_id)]
+      let result = Current_ocluster.build sched ~options ~pool:(Arch.pool_id arch) ~src:(Current.map (fun x -> [x]) commit) dockerfile in
+      let+ result = result
```

Comment:
This looks wrong. You want to record the build in the index without waiting for it to complete.

Reply:
Should this have an `Index.record` call without the job_id before and a second call to `record` with the job_id when we have one?

Reply:
You always have a job ID. What you don't always have is the job's result. You can just return the `result` `Current.t` directly here without waiting (or a tuple of that and the job ID current). Once you have all the jobs, you can pass that to the index code and, separately, wait for the results.

Reply:
@talex5 this is topical for me as well with the work I am doing on `ocaml-ci` -- `get_job_id` [returns an](https://github.com/ocurrent/ocurrent-deployer/pull/133/files#diff-5f0eea7390672da3c255b6d2e394932d240c9129a0db3eb95df34ed94a098a43R166) `option` and so I've had to work with the principle that we do not always have a job id. See also Have I misunderstood this?

Reply:
Ah, you're right: there might also be cases where you don't know the job ID, e.g. if the source hasn't been downloaded yet. That's not a problem, though - the index already accepts an option.

Reply:
@talex5 I'm interested in how `Index.record` could be used properly to support rebuilding jobs from GitHub actions. After far too much messing with build graphs I wasn't able to get them to display properly and function correctly.